sigal/gallery.py

```diff
@@ -521,37 +520,10 @@ def show_map(self):
 
     @cached_property
     def zip(self):
-        """Make a ZIP archive with all media files and return its path.
-
-        If the ``zip_gallery`` setting is set,it contains the location of a zip
-        archive with all original images of the corresponding directory.
-
+        """Placeholder ZIP method.
+        The ZIP logic is controlled by the zip_gallery plugin
         """
-        zip_gallery = self.settings['zip_gallery']
-
-        if zip_gallery and len(self) > 0:
-            zip_gallery = zip_gallery.format(album=self)
-            archive_path = join(self.dst_path, zip_gallery)
-            if (self.settings.get('zip_skip_if_exists', False) and
-                    isfile(archive_path)):
-                self.logger.debug("Archive %s already created, passing",
-                                  archive_path)
-                return zip_gallery
-
-            archive = zipfile.ZipFile(archive_path, 'w', allowZip64=True)
-            attr = ('src_path' if self.settings['zip_media_format'] == 'orig'
-                    else 'dst_path')
-
-            for p in self:
-                path = getattr(p, attr)
-                try:
-                    archive.write(path, os.path.split(path)[1])
-                except OSError as e:
-                    self.logger.warn('Failed to add %s to the ZIP: %s', p, e)
-
-            archive.close()
-            self.logger.debug('Created ZIP archive %s', archive_path)
-            return zip_gallery
+        return None
```

Comment:
I wonder if having the placeholder method is necessary? Jinja evaluates missing attributes to an empty string - I think, so it should be fine if the `.zip` method is missing.

Reply:
I don't think this is _necessary_. I just thought this can make the code _easier to read_.
For the reader, this _placeholder_ may give a hint on where the actual implementation of the _zip_ property is made. (I'm not a huge fan of _magic properties that just appears you don't know where from_ even if python allows it :smile: )

Anyway, If you'd like me to remove the placeholder, no problem for me. I can simply replace it with a comment to help future readers.

Reply:
I see your point about the magic aspect and agree in general, but on the other side it is weird to have some unrelated zip stuff here, since this is now an independent feature. And the implementation will be easy to find in the zip plugin.